File: backend/models/todo.go
package models

import "database/sql"

type Todo struct {
	TodoId    int          `json:"todo_id"`
	Content   string       `json:"content"`
	IsDone    bool         `json:"is_done"`
	DeletedAt sql.NullTime `json:"deleted_at"`
}

type ContentData struct {
	Content string `json:"content" validate:"required"`
}

File: frontend/src/app/entities/todo.ts
export type TodoType = {
    todo_id: number;
    content: string;
    is_done: boolean;
    deleted_at: string | null;
};

export class Todo {
    todoId: number;
    content: string;
    isDone: boolean;
    deletedAt: Date | null;

    constructor(properties?: TodoType) {
        this.todoId = properties ? properties.todo_id : 0;
        this.content = properties ? properties.content : '';
        this.isDone = properties ? properties.is_done : false;
        this.deletedAt = properties ? new Date(properties.deleted_at ?? "") : null;
    }
}
